modules/solid_mechanics/include/neml2/interfaces/NEML2ModelInterface.h

```diff
@@ -46,12 +52,27 @@ class NEML2ModelInterface : public T
   /// Get the target compute device
   const torch::Device & device() const { return _device; }
 
+  using RJType = std::tuple<neml2::ValueMap, neml2::DerivMap>;
+  using DispatcherType =
+      neml2::WorkDispatcher<neml2::ValueMap, RJType, RJType, neml2::ValueMap, RJType>;
+
+  /// Get the work scheduler
+  const std::shared_ptr<neml2::WorkScheduler> & scheduler() const { return _scheduler; }
+  /// Get the work dispatcher
+  const std::unique_ptr<DispatcherType> & dispatcher() const { return _dispatcher; }
+
 private:
+  /// The device on which to evaluate the NEML2 model
+  const torch::Device _device;
   /// The NEML2 material model
   neml2::Model & _model;
 
-  /// The device on which to evaluate the NEML2 model
-  const torch::Device _device;
+  /// The work scheduler to use
+  std::shared_ptr<neml2::WorkScheduler> _scheduler;
```

Comment:
Does this need to be a smart ptr? Isn't just the pointer ok?

If this remains a shared ptr:
```
const std::shared_ptr<neml2::WorkScheduler>
```
otherwise
```
neml2::WorkScheduler * const
```